File: app/src/main/java/com/github/ykiselev/playground/init/FrameInfoTracker.java
package com.github.ykiselev.playground.init;

import com.github.ykiselev.common.fps.FrameInfo;

import static org.lwjgl.glfw.GLFW.glfwGetTime;

public final class FrameInfoTracker extends FrameTask {

    private final FrameInfo frameInfo;

    private double t0;

    public FrameInfoTracker(Runnable next, FrameInfo frameInfo) {
        super(next);
        this.frameInfo = frameInfo;
    }

    @Override
    protected void onFrameStart() {
        t0 = glfwGetTime();
    }

    @Override
    protected void onFrameEnd() {
        final double t1 = glfwGetTime();
        frameInfo.add((t1 - t0) * 1000.0);
    }
}

File: app/src/main/kotlin/com/github/ykiselev/playground/init/FrameInfoTracker.kt
package com.github.ykiselev.playground.init

import com.github.ykiselev.common.fps.FrameInfo
import org.lwjgl.glfw.GLFW

class FrameInfoTracker(next: Runnable?, private val frameInfo: FrameInfo) : FrameTask(next) {

    private var t0 = 0.0

    override fun onFrameStart() {
        t0 = GLFW.glfwGetTime()
    }

    override fun onFrameEnd() {
        val t1 = GLFW.glfwGetTime()
        frameInfo.add((t1 - t0) * 1000.0)
    }
}
